client/src/components/TaskDetails.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+
+// Mapping between database column names and display subheading titles to reduce further data processesing
+const sectionNamesMap: { [key: string]: string } = {
+  category: "Task Category",
+  title: "Task Title",
+  created_at: "Date",
+  suburb: "Suburb",
+  state: "State",
+  estimated_time: "Estimated Time", // Missing ERD column name (fix later when db added)
+  budget: "Estimated Price",
+  description: "Task Details",
+};
+
+// Type of the section data to be passed
+type SectionData = {
+  [key: string]: string | null; // Allow database column to have null for item (prints 'N/A')
+};
+
+// Props for TaskDetails (and the sections)
+type TaskDetailsProps = {
+  data: SectionData;
+  emptyContentPlaceholder?: string; // OPTIONAL: What to show if missing content for a subheading.
+};
+
+const TaskDetailsSection = ({
+  data,
+  emptyContentPlaceholder = "N/A", // Change this to modify default placeholder
+}: TaskDetailsProps) => {
+  return (
+    <>
+      {Object.keys(sectionNamesMap).map((dbName, index) => (
+        <div
+          key={index}
+          className="py-2.5 text-base font-normal leading-5 text-[#485358]"
+        >
+          {sectionNamesMap[dbName]}
```

Comment:
By the way do you mind making this into its own `<Subheading data=.../>` component? So instead of displaying sectionNamesMap it displays any text, this is used by forms.